Client/Client.cc

```diff
@@ -158,6 +172,17 @@ TimeoutException::TimeoutException(const std::string& error)
 {
 }
 
+ConfigurationExceptionBad::ConfigurationExceptionBad(const std::string& error)
+    : Exception(error)
+{
+}
+
+ConfigurationExceptionChanged
+::ConfigurationExceptionChanged(const std::string& error)
```

Comment:
Style nit: keep the `ConfigurationExceptionChanged::ConfigurationExceptionChanged` together if you can. Break line after (, indent a couple of times.